lang/src/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransform.kt

```diff
@@ -257,21 +257,33 @@ internal class StaticTypeInferenceVisitorTransform(
         override fun transformExprPlus(node: PartiqlAst.Expr.Plus): PartiqlAst.Expr {
             val nAry = super.transformExprPlus(node) as PartiqlAst.Expr.Plus
             val type = when (nAry.operands.size) {
-                1    -> computeReturnTypeForArithmeticUnary(nAry, nAry.operands, "+")
+                1    -> throw IllegalArgumentException("PartiqlAst.Expr.Plus must have at least 2 arguments")
```

Comment:
nit: could change the `when` expression to be `nAry.operands.size < 2` to cover more cases in the error check

```
when (nAry.operands.size < 2) {
    true -> throw ...
    false -> computeReturnTypeForArithmeticNAry(...)
}
```